src/theme-manager.js

```diff
@@ -0,0 +1,401 @@
+/* global snapshotAuxiliaryData */
+
+const path = require('path')
+const _ = require('underscore-plus')
+const {Emitter, CompositeDisposable} = require('event-kit')
+const {File} = require('pathwatcher')
+const fs = require('fs-plus')
+const LessCompileCache = require('./less-compile-cache')
+
+// Extended: Handles loading and activating available themes.
+//
+// An instance of this class is always available as the `atom.themes` global.
+module.exports =
+class ThemeManager {
+  constructor ({packageManager, config, styleManager, notificationManager, viewRegistry}) {
+    this.packageManager = packageManager
+    this.config = config
+    this.styleManager = styleManager
+    this.notificationManager = notificationManager
+    this.viewRegistry = viewRegistry
+    this.emitter = new Emitter()
+    this.styleSheetDisposablesBySourcePath = {}
+    this.lessCache = null
+    this.initialLoadComplete = false
+    this.packageManager.registerPackageActivator(this, ['theme'])
+    this.packageManager.onDidActivateInitialPackages(() => {
+      this.onDidChangeActiveThemes(() => this.packageManager.reloadActivePackageStyleSheets())
+    })
+  }
+
+  initialize ({resourcePath, configDirPath, safeMode, devMode}) {
+    this.resourcePath = resourcePath
+    this.configDirPath = configDirPath
+    this.safeMode = safeMode
+    this.lessSourcesByRelativeFilePath = null
+    if (devMode || (typeof snapshotAuxiliaryData === 'undefined')) {
```

Comment:
@as-cii: The linter was complaining about `snapshotAuxiliaryData` not being defined. c06745f098a2c462c3bacc569b4aa53c868ed62b tells the linter that `snapshotAuxiliaryData` is a global variable, and that suppresses the linter error. Given your work in this area of the code in 4c6805bf7647efdf2d5d607e0b6dea1f98e9ed73, does this seem like a reasonable approach to dealing with the linter error?

Reply:
Yep, that looks good! Another approach would be to move that global into `package.json`, where I believe we define other globals not recognized by the linter.

Reply:
> Another approach would be to move that global into package.json, where I believe we define other globals not recognized by the linter.

Thanks, @as-cii. Based on the search below, it looks like `snapshotAuxiliaryData` is only referenced as a global in this one file, so I'll whitelist it just in this file. Thanks for the help! :bow:

```
$ ag snapshotAuxiliaryData
script/config.js
40:  snapshotAuxiliaryData: {}

script/lib/prebuild-less-cache.js
32:  CONFIG.snapshotAuxiliaryData.lessSourcesByRelativeFilePath = {}
35:    if (!CONFIG.snapshotAuxiliaryData.lessSourcesByRelativeFilePath.hasOwnProperty(relativeFilePath)) {
36:      CONFIG.snapshotAuxiliaryData.lessSourcesByRelativeFilePath[relativeFilePath] = {
43:  CONFIG.snapshotAuxiliaryData.importedFilePathsByRelativeImportPath = {}
65:        if (!CONFIG.snapshotAuxiliaryData.importedFilePathsByRelativeImportPath.hasOwnProperty(relativeImportPath)) {
66:          CONFIG.snapshotAuxiliaryData.importedFilePathsByRelativeImportPath[relativeImportPath] = []
68:            CONFIG.snapshotAuxiliaryData.importedFilePathsByRelativeImportPath[relativeImportPath].push(

script/lib/generate-startup-snapshot.js
17:    auxiliaryData: CONFIG.snapshotAuxiliaryData,

src/theme-manager.js
1:/* global snapshotAuxiliaryData */
36:    if (devMode || (typeof snapshotAuxiliaryData === 'undefined')) {
40:      this.lessSourcesByRelativeFilePath = snapshotAuxiliaryData.lessSourcesByRelativeFilePath
41:      this.importedFilePathsByRelativeImportPath = snapshotAuxiliaryData.importedFilePathsByRelativeImportPath
```